themes/classic/templates/catalog/_partials/miniatures/product.tpl

```diff
@@ -69,7 +69,15 @@
 
               {hook h='displayProductPriceBlock' product=$product type="before_price"}
 
-              <span class="price" aria-label="{l s='Price' d='Shop.Theme.Catalog'}">{$product.price}</span>
+              <span class="price" aria-label="{l s='Price' d='Shop.Theme.Catalog'}">
+                {capture name='custom_price'}{hook h='displayProductPriceBlock' product=$product type='custom_price' hook_origin='products_list'}{/capture}
+                {if $smarty.capture.custom_price}
```

Comment:
Question 😛 I'm being a bit paranoid but what happens if `0` is returned (so, following your example, the module wants to display price zero as "free !" instead of regular price), this `if` will behave correctly ?

Maybe use
```suggestion
                {if $smarty.capture.custom_price ne ""}
```
to be sure ? 😅

Reply:
Hi @matks 
In this case a programmer will make a mistake if he allows a value such as '0' to be returned. Here we expect a non-empty string to be returned, that is, a formatted number or some another text. In my approach, '0' will not be displayed.
Perhaps a main question here is whether you have a rule in PrestaShop guideline about whether it is imperative to always avoid implicit comparison. If there is no such rule, then it might be worth doing ;) It's not hard for me to make an explicit comparison, which I did now.